src/coreclr/jit/codegenxarch.cpp

```diff
@@ -700,6 +735,17 @@ void CodeGen::genSetRegToConst(regNumber targetReg, var_types targetType, GenTre
             break;
         }
 
+        case GT_CNS_MSK:
+        {
+#if defined(FEATURE_MASKED_HW_INTRINSICS)
+            GenTreeMskCon* mskCon = tree->AsMskCon();
+            genSetRegToConst(mskCon->GetRegNum(), targetType, &mskCon->gtSimdMaskVal);
+#else
+            unreached();
```

Comment:
at multiple places, for `GT_CNS_MSK` we have 

```
#if defined(FEATURE_MASKED_HW_INTRINSICS)
   // logic
#else
   unreached();
#endif
```

wondering if there is a single place where we can add this check (for example, during the creation of such node) and then assume that we will never have `GT_CNS_MSK` when `FEATURE_MASKED_HW_INTRINSICS` is not enabled? I think this goes with my earlier comment if we can just have the definition of `GT_CNS_MSK` under `#ifdef FEATURE_MASKED_HW_INTRINSICS`

Reply:
Yes, that may be better overall. I'll investigate it in a follow up PR as indicated above.